src/openforms/emails/signals.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from django_yubin.models import Message
+
+from openforms.logging import logevent
+from openforms.submissions.models import Submission
+
+from .constants import (
+    X_OF_CONTENT_TYPE_HEADER,
+    X_OF_CONTENT_UUID_HEADER,
+    X_OF_EVENT_HEADER,
+    EmaiContentTypeChoices,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=Message)
+def yubin_messages_status_change_handler(signal, sender, instance, created, **kwargs):
+    if created:
+        return
+
+    message = instance.get_email_message()
+    has_submission = (
+        message.extra_headers.get(X_OF_CONTENT_TYPE_HEADER, "")
+        == EmaiContentTypeChoices.submission
+    )
+    if not has_submission:
+        return
+
+    submission_uuid = message.extra_headers.get(X_OF_CONTENT_UUID_HEADER)
+    event = message.extra_headers.get(X_OF_EVENT_HEADER)
+    if not submission_uuid:
+        return
```

Comment:
there is missing coverage here, but rather than making up a false example where you would only include part of the e-mail messages, I think an `assert submission_uuid` makes more sense and should crash/error the tests when someone forgets to add the actual submission UUID header to an email.